### PR TITLE
Set the surname of the test advocate

### PR DESCRIPTION
**Pull request by jrmhaig**
#### What

Fix one of the flickering tests.

#### Ticket

N/A

#### Why

One of the tests (spec/controllers/external_users/claims_controller_spec.rb:475) flickers because it is searching for the advocates surname and expecting nothing to be returned but the surname is randomly chosen and 'Smith' is used as a defendant name.

#### How

The solution is to fix the name of the advocate to ensure that a result is not found.

### CODE REVIEW

File: spec/factories/external_users.rb [hunk 1]
```diff
@@ -17,11 +17,12 @@
   factory :external_user do
     transient do
       build_user { true }
+      last_name { Faker::Name.last_name }
     end
 
     after(:build) do |eu, evaluator|
       if evaluator.build_user
-        eu.user ||= build(:user, first_name: Faker::Name.first_name, last_name: Faker::Name.last_name, email: Faker::Internet.email, password: 'password', password_confirmation: 'password')
+        eu.user ||= build(:user, first_name: Faker::Name.first_name, last_name: evaluator.last_name, email: Faker::Internet.email, password: 'password', password_confirmation: 'password')
```
Comment (jsugarman) [inline spec/factories/external_users.rb]: will this not need to fallback if the evaluator has not supplied a last name
```suggestion
        eu.user ||= build(:user, first_name: Faker::Name.first_name, last_name: evaluator.last_name || Faker::Name.last_name, email: Faker::Internet.email, password: 'password', password_confirmation: 'password')
```

Comment (jrmhaig) [reply spec/factories/external_users.rb]: If this is not set in the `create` or `build` then it gets set on line 20, above.

File: spec/controllers/external_users/claims_controller_spec.rb [hunk 2]
```diff
@@ -448,28 +469,26 @@ def advocate
...
-        @allocated_claim = create(:allocated_claim, external_user: advocate)
-        create(:defendant, claim: @allocated_claim, first_name: 'Pete', last_name: 'Adams')
+        create(:defendant, claim: archived_claim, first_name: 'John', last_name: 'Smith')
+        create(:defendant, claim: draft_claim, first_name: 'John', last_name: 'Smith')
+        create(:defendant, claim: allocated_claim, first_name: 'Pete', last_name: 'Adams')
       end
 
-      context 'in all claims' do
-        context 'by defendant name' do
+      describe  'in all claims' do
```
Comment (jsugarman) [inline spec/controllers/external_users/claims_controller_spec.rb]: `describe` should be for a method  `#method_name|.class_method_name`

File: spec/controllers/external_users/claims_controller_spec.rb [hunk 3]
```diff
@@ -482,21 +501,21 @@ def advocate
...
-              expect(assigns(:claims).sort_by(&:state)).to eq([@allocated_claim, @draft_claim])
+              expect(assigns(:claims).sort_by(&:state)).to eq([allocated_claim, draft_claim])
             end
           end
         end
       end
 
-      context 'in archive' do
-        context 'by defendant name' do
+      describe 'in archive' do
```
Comment (jsugarman) [inline spec/controllers/external_users/claims_controller_spec.rb]: describe should be for a method #method_name|.class_method_name

File: spec/factories/external_users.rb (hunk 1 above)
Comment (jsugarman) [reply spec/factories/external_users.rb]: right! i'd be tempted to go further then and make the first_name, and even email transient with default, as for last name. would be more consistent and less confusing.

**Review by jsugarman — CHANGES_REQUESTED** [review]

File: spec/factories/external_users.rb (hunk 1 above)
Comment (jrmhaig) [reply spec/factories/external_users.rb]: I could do this. I was going for the minimum possible change but it makes sense to be able to set the other fields too.

Comment (jrmhaig) [reply spec/factories/external_users.rb]: Actually, I had second thought and made it simpler with bb6347a86908a8e1c86d868bca1065203f7228a5

**Review by jsugarman — APPROVED** [review]
👍

**Pull request merged**